src/include/catalog/index_schema.h

```diff
@@ -419,6 +421,7 @@ class IndexSchema {
 
   friend class Catalog;
   friend class postgres::Builder;
+  friend class postgres::PgCoreImpl;
```

Comment:
nit: Can you consolidate this cluster with `friend class DatabaseCatalog;` at the start of the `private` section (exactly where they sit doesn't matter so much, but they should be together)?

Reply:
moved